### DISCUSSION TITLE
how to package a .py into .exe ?

### DISCUSSION THREAD

**Discussion by Aditelte** [TomSchimansky/CustomTkinter#1175] [General]

**Comment by Tenimes** [comment]
You need to install PyInstaller (Auto py to Exe) first. After installing you have to open terminal and cd the directory where your code is. Then you have to tpye this command: `pyinstaller --noconfirm --onedir --windowed --add-data "<CustomTkinter Location>/customtkinter;customtkinter/"  "<Path to Python Script>"`. You can find more details at [https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging](url)

**Comment by Wolf-SO** [comment]
> You need to install PyInstaller (Auto py to Exe) first. After installing you have to open terminal and cd the directory where your code is. Then you have to tpye this command: `pyinstaller --noconfirm --onedir --windowed --add-data "<CustomTkinter Location>/customtkinter;customtkinter/" "<Path to Python Script>"`. You can find more details at `https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging`

For some reason, the above link is broken, even if it looks right. For those interested, the following one will do meanwhile:
https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging

**Comment by Wolf-SO** [comment]
@Aditelte For questions like this, **Discussions** is a better place than **Issues**. 

There is also a search function. See for example the results for [`executable`](https://github.com/TomSchimansky/CustomTkinter/discussions?discussions_q=executable)

**Comment by Akascape** [comment]
A full detailed guide is given here: https://github.com/TomSchimansky/CustomTkinter/discussions/939
